aws/resource_aws_db_proxy_test.go

```diff
@@ -0,0 +1,274 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_db_proxy", &resource.Sweeper{
+		Name: "aws_db_proxy",
+		F:    testSweepRdsDbProxies,
+	})
+}
+
+func testSweepRdsDbProxies(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("Error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).rdsconn
+
+	err = conn.DescribeDBProxiesPages(&rds.DescribeDBProxiesInput{}, func(out *rds.DescribeDBProxiesOutput, lastPage bool) bool {
+		for _, dbpg := range out.DBProxies {
+			if dbpg == nil {
+				continue
+			}
+
+			input := &rds.DeleteDBProxyInput{
+				DBProxyName: dbpg.DBProxyName,
+			}
+			name := aws.StringValue(dbpg.DBProxyName)
+
+			log.Printf("[INFO] Deleting DB Proxy: %s", name)
+
+			_, err := conn.DeleteDBProxy(input)
+
+			if err != nil {
+				log.Printf("[ERROR] Failed to delete DB Proxy %s: %s", name, err)
+				continue
+			}
+		}
+
+		return !lastPage
+	})
+
+	if testSweepSkipSweepError(err) {
+		log.Printf("[WARN] Skipping RDS DB Proxy sweep for %s: %s", region, err)
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("Error retrieving DB Proxies: %s", err)
+	}
+
+	return nil
+}
+
+func TestAccAWSDBProxy_basic(t *testing.T) {
+	var v rds.DBProxy
+	resourceName := "aws_db_proxy.test"
+	name := fmt.Sprintf("tf-acc-db-proxy-%d", acctest.RandInt())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSDBProxyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSDBProxyConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSDBProxyExists(resourceName, &v),
+					resource.TestCheckResourceAttr(
+						resourceName, "name", name),
+					resource.TestCheckResourceAttr(
+						resourceName, "engine_family", "MYSQL"),
+					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "rds", regexp.MustCompile(`db-proxy:.+`)),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
```

Comment:
New resources should include `_disappears` testing, following the [Contributing Guide](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md#disappears-acceptance-tests):

e.g.

```suggestion

func TestAccAWSDBProxy_disappears(t *testing.T) {
	var v rds.DBProxy
	resourceName := "aws_db_proxy.test"
	name := fmt.Sprintf("tf-acc-db-proxy-%d", acctest.RandInt())

	resource.ParallelTest(t, resource.TestCase{
		PreCheck:     func() { testAccPreCheck(t) },
		Providers:    testAccProviders,
		CheckDestroy: testAccCheckAWSDBProxyDestroy,
		Steps: []resource.TestStep{
			{
				Config: testAccAWSDBProxyConfig(name),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckAWSDBProxyExists(resourceName, &v),
					testAccCheckResourceDisappears(testAccProvider, resourceAwsDbProxy(), resourceName),
				),
				ExpectNonEmptyPlan: true,
			},
		},
	})
}

```

Reply:
Looks like this was handled, so resolving the discussion.